app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/GnosisSafeTransactionRepository.kt

```diff
@@ -173,17 +176,43 @@ class GnosisSafeTransactionRepository @Inject constructor(
             calculateHash(safeAddress, transaction).flatMap {
                 Single.fromCallable {
                     val transactionUuid = UUID.randomUUID().toString()
-                    descriptionsDao.insertDescription(
+                    descriptionsDao.insert(
                             TransactionDescriptionDb(transactionUuid, safeAddress, transaction.address,
                                     transaction.value?.value ?: BigInteger.ZERO,
                                     transaction.data ?: "", DEFAULT_OPERATION,
                                     transaction.nonce ?: DEFAULT_NONCE, System.currentTimeMillis(), null, it.toHexString())
-
+                    )
+                    descriptionsDao.insert(
+                            TransactionPublishStatusDb(transactionUuid, txChainHash, null)
                     )
```

Comment:
Maybe we can do both `inserts` in the context of 1 Transaction by using [`@Transaction`](https://developer.android.com/reference/android/arch/persistence/room/Transaction.html)

Reply:
we would have to define that on the Dao, right? We could do that, but it feels a little weird. Let's check for stuff like that later :)

Reply:
Yeah it needs to be in the Dao 😉